packages/playwright-core/src/server/injected/roleUtils.ts

```diff
@@ -461,6 +461,38 @@ export function getElementAccessibleDescription(element: Element, includeHidden:
   return accessibleDescription;
 }
 
+export function getElementAccessibleErrorMessage(element: Element, includeHidden: boolean): string {
+  const cache = includeHidden ? cacheAccessibleErrorMessageHidden : cacheAccessibleErrorMessage;
+  let accessibleErrorMessage = cache?.get(element);
+
+  if (accessibleErrorMessage === undefined) {
+    accessibleErrorMessage = '';
+
+    const ariaInvalid = element.getAttribute('aria-invalid');
+    if (ariaInvalid === 'true') {
```

Comment:
The spec says "Any value not recognized in the list of allowed values MUST be treated by user agents as if the value true had been provided.", should we take this into account here?

Reply:
```ts
const ariaInvalid = element.getAttribute('aria-invalid');
const isAriaInvalid = ariaInvalid !== null && ariaInvalid.toLowerCase() !== 'false';
```
After making the code changes, I added [regression tests](https://github.com/microsoft/playwright/pull/33904/commits/1f6d9ca77d73ebacbc0de9ae38e2cc9472f4de2b).
If there's anything I missed, I would appreciate your feedback.